docs/fobnail-api.md

```diff
@@ -0,0 +1,387 @@
+section below).
+
+Fobnail Token verifies the provided metadata against AIK bound to the
+Provisioning Context, if signature verification fails **4.03** is returned, if
+metadata format is invalid **4.00** is returned. On success either **2.01** or
+**2.04**. On the first successful call **2.01** is returned (`Location-Path`
+must not be present), on subsequent calls **2.04** is returned to signify that
+metadata has been replaced.
+
+#### /admin/provision/{id}/rim
+
+RIM holds PCR registers (from TPM) which are used for attestation. It doesn't
+contain information about the Attester since these are part of metadata. RIM
+may contain multiple PCR banks (depending of hash algorithm: SHA-1, SHA-256,
+others). Each PCR bank contains bitmap of present PCRs (`pcrs` field) and array
+of PCR itself. The LSB of bitmap corresponds to PCR0. RIMs are sent in CBOR:
+
+```json
+{
+    // PCR update counter, incremented every time when some PCR gets updated.
+    "update_ctr": 0,
+    // Array of PCR banks (major 4)
+    "banks": [
+        // First PCR bank
+        {
+            // Algorithm ID as used by TPMs (TPM2_ALG_* constants from libtss)
+            // (major 0, unsigned integer)
+            "algo_id": 0xb,
+            // Bitmap of present PCRs (major 0, unsigned integer)
+            "pcrs": 0xffffffff,
+            // Array of PCR registers, each register holds hash corresponding to the
+            // type of bank.
+            // There must be as many hashes as set bits in "pcrs" field. If
+            // these don't match then RIM is considered invalid.
+            "pcr": [
+                // Size of hash is verified, whether all hashes in a bank have
+                // the same size, if these don't match then entire RIM is
+                // considered invalid
+                [ 0x00, 0x00, 0x00, 0x00, ... ],
+                ...
+            ]
+        },
+        // Another PCR bank (major 5, map)
+        {
+            "algo_id": 0xb,
+            "pcrs": 0x000000ff,
+            // major 4 (array)
+            "pcr": [
+                // major 2 (byte string)
+                [ 0x00, 0x00, 0x00, 0x00, ... ],
+                ...
+            ]
+        }
+    ],
+}
+```
+
+Client **must** sign RIM with AIK (see [Data signing](#data-signing) section
+below).
+
+Fobnail Token verifies RIM signature and format, if signature verification fails
+**4.03** is returned, if RIM has invalid format **4.00** is returned. On success
+**2.01** or **2.04** is returned. On the first successful call **2.01** is
+returned (`Location-Path` must not be present), on subsequent calls **2.04** is
+returned to signify that the previous RIM has been replaced with the new RIM.
+
+#### /admin/provision/{id}
+
+POST request to this endpoint results in writing data attached to the
+Provisioning Context into persistent storage. Currently, this request does not
+contain any data and payload **must** be empty and or **4.00** response will be
```

Comment:
```suggestion
contain any data and payload **must** be empty or **4.00** response will be
```

Also, do we expect any payload in the future? i think if there were anything to be passed to Token, it should have a dedicated API.

Reply:
Fixed